### DISCUSSION TITLE
dde 项目应该如何 debug？

### DISCUSSION THREAD

**Discussion by chunibyo-wly** [linuxdeepin/developer-center#3559] [Q&A | 问答板块] [answered]
请问我想使用 qtcreator debug dde-file-manager, 但是每次在界面初始化完成后 debug 进程都会退出，应该如何解决?

1. 断点正常使用
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/33208530/215343253-3265fe11-7a2e-4e56-8a98-4503ad175ee0.png">

2. 界面初始化完成后，断点不生效
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/33208530/215343354-f4e37cae-e59a-4493-884e-a00edf869daf.png">

**Comment by BLumia** [comment, marked as answer]
dde-file-manager 应该是可以正常 debug 的才对，按你的描述的话，可能可以检查下面几点：

1. 启动的目标是不是 dde-file-manager。默认可能会选到 dde-desktop，入口文件不是同一个。
2. 检查你在调试之前是否将所有的文管窗口关闭。文管我印象里是单实例进程，如果有已经存在的进程就不会启动新的，会让原有进程开新的窗口，而新的则直接退出掉。

顺便，我们有个面向开发者的 [Matrix 实时聊天频道](https://wiki.deepin.org/Matrix)，可以考虑加进来讨论 :)

**Comment by chunibyo-wly** [comment]
您说的对，确实有残留进程，问题解决了，非常感谢
